src/main.ts

```diff
@@ -95,100 +102,168 @@ export default class AdvancedURI extends Plugin {
         });
 
 
-        this.registerObsidianProtocolHandler("advanced-uri", async (e) => {
-            const parameters = e as unknown as Parameters;
+        this.registerObsidianProtocolHandler("advanced-uri", this.handleUriAction);
+
+        this.registerEvent(
```

Comment:
This _looks_ like an add, but it's actually just a confusing diff; it's already present in the existing code, it's just further down below the moved `handleUriAction` content.